### PR TITLE
[FIX] sale_timesheet: performance for project with a lot of tasks

### PR DESCRIPTION
**Pull request by tbe-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_timesheet/models/project_overview.py [hunk 1]
```diff
@@ -383,9 +398,13 @@ def _plan_get_stat_button(self):
         # if only one project, add it in the context as default value
         tasks_domain = [('project_id', 'in', self.ids)]
         tasks_context = self.env.context
-        tasks_projects = self.env['project.task'].sudo().search(tasks_domain).mapped('project_id')
-        if len(tasks_projects) == 1:
-            tasks_context = {**tasks_context, 'default_project_id': tasks_projects.id}
+
+        # filter out all the projects that have no tasks
+        task_projects_ids = self.env['project.task'].read_group([('project_id', 'in', self.ids)], ['project_id'], ['project_id'])
```
Comment (nseinlet) [inline addons/sale_timesheet/models/project_overview.py]: This line demonstrate you know the read Group ;-)

File: addons/sale_timesheet/models/project_overview.py [hunk 2]
```diff
@@ -383,9 +398,13 @@ def _plan_get_stat_button(self):
...
+        task_projects_ids = [p['project_id'][0] for p in task_projects_ids]
+
+        if len(task_projects_ids) == 1:
```
Comment (nseinlet) [inline addons/sale_timesheet/models/project_overview.py]: search_count ?

Comment (tbe-odoo) [reply addons/sale_timesheet/models/project_overview.py]: Unfortunately, if `task_projects_ids` has only one cell, then we use its value :/

**Comment by tbe-odoo** [comment]
Thanks for the review @nseinlet :)

May I r+ ?

**Comment by tbe-odoo** [comment]
Coucou @rco-odoo , je pense que tu as sûrement déjà eu ce feedback, mais ce serait bien que l'on puisse faire des `search` qui font des jointures sans pour autant avoir d'`auto_join` sur le champs relationnel.

Sur cette PR qui tente d'améliorer les performances, ça aurait un bel effet (et le code serait plus propre :P)

**Comment by nseinlet** [comment]
@robodoo  r+

**Pull request closed without merging**